### ISSUE TITLE
problem on unordered_map<string,vector<T>>

### ISSUE THREAD

**Issue by sunqi1993** [jorgen/json_struct#10]
seems that `unordered_map<string,vector<T>>` is not suupported
the  key of unordered_map  is empty string

test case
```
#include <string>
#include <vector>
#include <map>
#include <json_struct.h>
#include <cinttypes>
//these two have to be ifdef guarded becuase JS support compiler versions where
//they are not implemented
#ifdef JS_STD_UNORDERED_MAP
#include <unordered_map>
#endif
#ifdef JS_STD_OPTIONAL
#include <optional>
#endif


const char json[] = R"json(
{
    "unordered_map" : {
        "foo" : [1],
        "bar" : [2]
    }
}
)json";

struct JsonData
{
#ifdef JS_STD_UNORDERED_MAP
    std::unordered_map<std::string, std::vector<double>> unordered_map;
#else
    JS::JsonObject unordered_map;
#endif
  JS_OBJ(unordered_map);

};

int main()
{
    JsonData dataStruct;
    JS::ParseContext parseContext(json);
    if (parseContext.parseTo(dataStruct) != JS::Error::NoError)
    {
        std::string errorStr = parseContext.makeErrorString();
        fprintf(stderr, "Error parsing struct %s\n", errorStr.c_str());
        return -1;
    }

#ifdef JS_STD_UNORDERED_MAP
    fprintf(stderr, "unordered_map:\n");
    for (auto it : dataStruct.unordered_map)
        fprintf(stderr, "\t%s", it.first.c_str());  // empty string here
#endif
    return 0;
}
```

**Comment by jorgen** [comment]
sorry about the lack of response. I will fix it this afternoon. The idea about the ifdefs is that if the user application supports unordered_map then they should be used unconditionally.

**Comment by jorgen** [comment]
Thank you for your bug report. This is now fixed by change: 4dfd01b.

**Comment by sunqi1993** [comment]
thanks for your reply and for your lib, this library is awaresaome and save us a lot time @jorgen
